org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/hover/ScalaHoverImpl.scala

```diff
@@ -116,7 +116,21 @@ class ScalaHoverImpl extends ITextHover with ITextHoverExtension with ITextHover
 
         val scalaRegion = new Region(icu.sourceMap(viewer.getDocument.get.toCharArray).scalaPos(region.getOffset), region.getLength)
 
-        def typeInfo(t: Tree): Option[String] = (for (sym <- Option(t.symbol); tpe <- Option(t.tpe)) yield compiler.headerForSymbol(sym, tpe)).flatten
+        def typeInfo(t: Tree): Option[String] = (for (sym <- Option(t.symbol); tpe <- Option(t.tpe)) yield {
+          if( sym.isMethod && sym.isSourceMethod ){//this is info message for 'var's. We check if owner has both setter and getter
+            val gsym = sym.getterIn(sym.owner)
+            val ssym = sym.setterIn(sym.owner)
+            if( gsym != NoSymbol && ssym != NoSymbol)
```

Comment:
This nested if can be removed by converting `gsym` and `ssym` to a def and then call them in the first if.